src/vm/methodtablebuilder.cpp

```diff
@@ -8832,10 +8832,13 @@ void MethodTableBuilder::CopyExactParentSlots(MethodTable *pMT, MethodTable *pAp
         // just the first indirection to detect sharing.
         if (pMT->GetVtableIndirections()[0].GetValueMaybeNull() != pCanonMT->GetVtableIndirections()[0].GetValueMaybeNull())
         {
+            MethodTable::MethodDataWrapper hCanonMTData(MethodTable::GetMethodData(pCanonMT, FALSE));
             for (DWORD i = 0; i < nParentVirtuals; i++)
             {
-                PCODE target = pCanonMT->GetRestoredSlot(i);
-                pMT->SetSlot(i, target);
+                MethodDesc *pMD = hCanonMTData->GetImplMethodDesc(i);
+                CONSISTENCY_CHECK(CheckPointer(pMD));
+                CONSISTENCY_CHECK(pMD == pCanonMT->GetMethodDescForSlot(i));
+                pMT->SetSlot(i, pMD->GetInitialEntryPointForCopiedSlot());
```

Comment:
Please break this out into a helper function. I see it copied in at least 4 places in the codebase.